str_analysis/call_non_ref_pathogenic_motifs.py

```diff
@@ -0,0 +1,978 @@
+
+        #locus_results_json[f"expansion_hunter_denovo_profile"] = args.expansion_hunter_denovo_profile
+        locus_results_json["ehdn_sample_read_depth"] = sample_read_depth
+
+        # get the 2 motifs with the most read support
+        records.sort(key=lambda r: (
+            -r["anchored_irr_count_for_this_repeat_unit_and_region"],
+            -r["total_irr_count_for_this_repeat_unit_and_region"],
+            r["repeat_unit"],
+        ))
+
+        for i in 0, 1:
+            if i >= len(records):
+                continue
+
+            record = records[i]
+            motif_number = i + 1
+            locus_results_json.update({
+                f"ehdn_motif{motif_number}_repeat_unit": record.get("repeat_unit"),
+                f"ehdn_motif{motif_number}_anchored_irr_count": record.get("anchored_irr_count_for_this_repeat_unit_and_region"),
+                f"ehdn_motif{motif_number}_n_anchored_regions": record.get("n_anchored_regions_for_this_repeat_unit"),
+                f"ehdn_motif{motif_number}_paired_irr_count": record.get("paired_irr_count_for_this_repeat_unit"),
+                f"ehdn_motif{motif_number}_total_irr_count": record.get("total_irr_count_for_this_repeat_unit_and_region"),
+            })
+
+    # generate output json
+    output_filename = f"{args.output_prefix}.{locus_id}_motifs.json"
+    with open(output_filename, "wt") as f:
+        json.dump(locus_results_json, f, indent=2)
+    print(f"Wrote results to {output_filename}")
+    pprint(locus_results_json)
+
+
+def compute_sample_id(bam_or_cram_path, reference_fasta):
+    """Determine a sample id from the given bam or cram file
+
+    Args:
+        bam_or_cram_path (str): bam or cram file path
+        reference_fasta (str): reference fasta file path
+    Return:
+        str: sample id
+    """
+
+    sample_id = None
+    # try to get sample id from bam/cram header
+    with pysam.Samfile(bam_or_cram_path, reference_filename=reference_fasta) as f:
+
+        read_groups = f.header.as_dict().get("RG", [])
+        if read_groups:
+            sample_id = read_groups[0].get("SM")
+
+    if sample_id:
+        print(f"Using sample id '{sample_id}' from the bam/cram header")
+    else:
+        sample_id = re.sub(".bam$|.cram$", "", os.path.basename(bam_or_cram_path))
+        print(f"Using sample id '{sample_id}' based on the input filename prefix")
+
+    return sample_id
+
+
+def main():
+    """Call non-ref pathogenic motifs"""
+
+    args = parse_args()
+
+    if not args.sample_id:
+        args.sample_id = compute_sample_id(args.bam_or_cram_path, args.reference_fasta)
+
+    args.output_prefix = args.output_prefix or args.sample_id
+
+    if args.run_expansion_hunter_denovo:
+        args.expansion_hunter_denovo_profile = run_expansion_hunter_denovo(args)
+
+    if args.locus:
+        loci = args.locus
+    elif args.all_loci:
+        loci = LOCUS_INFO.keys()
+    else:
+        raise ValueError("Must specify --locus or --all-loci")
```

Comment:
I tend to include else statements to catch logic errors and unexpected conditions in the future - for example, in case I change the arg spec later and forget to update this if statement.